src/kOS/AddOns/AddonManager.cs

```diff
@@ -0,0 +1,41 @@
+﻿using kOS.Safe.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace kOS.AddOns
+{
+    [AssemblyWalk(AttributeType = typeof(kOSAddonAttribute), InherritedType = typeof(Suffixed.Addon), StaticRegisterMethod = "RegisterMethod")]
+    public class AddonManager
+    {
+        public Dictionary<string, Suffixed.Addon> AllAddons { get; private set; }
+        private static readonly Dictionary<kOSAddonAttribute, Type> rawAttributes = new Dictionary<kOSAddonAttribute, Type>();
+        private readonly SharedObjects shared;
+
+        public AddonManager(SharedObjects sharedObj)
+        {
+            AllAddons = new Dictionary<string, Suffixed.Addon>(StringComparer.OrdinalIgnoreCase);
+            foreach (kOSAddonAttribute attr in rawAttributes.Keys)
+            {
+                AllAddons.Add(attr.Identifier, (Suffixed.Addon)Activator.CreateInstance(rawAttributes[attr], sharedObj));
+            }
```

Comment:
`AddonManager` is not instantiated until the `kOSProcessor`'s `InitializeObjects` method is called.  `rawAttributes` is filled by `RegisterMethod` during the initial assembly walk at the very beginning when KSP launches.  So the order goes: